capdpa/IR/ir_type.py

```diff
@@ -12,6 +12,12 @@ def __init__(self, name, pointer = 0, constant = False, reference = False, **kwa
         self.reference = reference
         self.constant = constant
 
+    def UsedTypes(self, parent):
+        fqn = self.FullyQualifiedName()
+        if parent == fqn[:-1] or (parent == fqn[:-2] and fqn[-1] == "Class"):
+            return []
+        return [self.FullyQualifiedName()]
```

Comment:
I'd put the second return in an else block:
```Python
    if parent == fqn[:-1] or (parent == fqn[:-2] and fqn[-1] == "Class"):
        return []
    else
        return [self.FullyQualifiedName()]
```
I think this is more clear (especially if the function grows). There was a talk about this matter, this was the preferred solution and I'd go with the argumentation they made (I can't rephrase it but I'll link the talk when I find it again).
I'm aware that we probably don't follow this style in our code base but I'd advocate to do so.